plugins/modules/iam_user.py

```diff
@@ -543,7 +580,12 @@ def main():
         mutually_exclusive=[['password', 'remove_password']],
     )
 
-    connection = module.client('iam')
+    module.deprecate("The 'iam_user' return key is deprecated and will be replaced by 'user'. Both values are returned for now.",
+                     date='2022-12-01', collection_name='community.aws')
```

Comment:
Does this have to be a date ? Would it be better to be explicit on the target release ?

Reply:
hm, I think it made in this way in the past too.  
See `grep -rn 2022-06-01 *`  
If I extrapolate...
  * 2022-06-01 will be 4.0.0
  * 2022-12-01 might be 5.0.0

Reply:
Yep, just probably makes it easier to keep track of if explicit and easier to prepare for the releases

Reply:
This is less than a year for deprecation, that isn't long enough.  Please set this two years out.